src/o365/spfx/commands/project/project-externalize/rules/PnPJsRule.ts

```diff
@@ -0,0 +1,21 @@
+import { BasicDependencyRule } from "./BasicDependencyRule";
+import { Project } from "../../project-upgrade/model";
+import { ExternalizeEntry } from "../model/ExternalizeEntry";
+
+export class PnPJsRule extends BasicDependencyRule {
+  public get ModuleName () {
+    return '@pnp/pnpjs';
+  }  
+  public visit(project: Project, findings: ExternalizeEntry[]): void {
+    if(project.packageJson) {
```

Comment:
Does it worth checking the externals in the config.json file before printing this?

We usually call the externals the same as the module name (not sure if this is a good practice) so the externals might already have it like:
```
 "externals": {
    "@pnp/pnpjs": {
      "path": "https://...",
      "globalName": "@pnp/pnpjs"
    }
```
or if they previously run the same command they will have

```
"externals": {
    "@pnp/pnpjs": {
      "path": "https://cdnjs.cloudflare.com/ajax/libs/pnp-pnpjs/${version}/pnpjs.es5.umd.bundle.min.js",
      "globalName": "pnp"
    }
```

Reply:
I'm not sure I understand the comment correctly.
Is the question: "should we check they don't already have an entry for the module we're recommending in the config.json so we don't run into the risk of duplicating entries?" => probably for a next version yes.
If the question is around the global name or something else, can you elaborate please? Thanks!

Reply:
Since we already have the whole project object accessible in code, I'm with Velin, that we should check if the particular dependency is not already listed in externals before reporting it.